test/set.test.js

```diff
@@ -0,0 +1,173 @@
+var test = require('tape');
+
+[['builtIn-js', Set], ['node-native-set', require('../index.js')]].forEach(([setType, Set]) => {
+  test(`test ${setType} constructor`, (t) => {
+    t.doesNotThrow(() => {new Set()}, 'can construct an empty new Set()');
+    t.doesNotThrow(() => {new Set(null)}, 'can construct with null');
+    t.doesNotThrow(() => {new Set([1,2,3]), 'can construct an iterable array of integers'});
+
+    let stringy;
+    t.doesNotThrow(() => {
+      stringy = new Set('stringy');
+    }, 'can construct set with string');
+    t.equal(stringy.size, 7, 'string set size is as expected');
+
+    let arrayOfArrays;
+    t.doesNotThrow(() => {
+      arrayOfArrays = new Set([[1,2], [3,4]]);
+    }, 'can construct an array of arrays');
+    t.equal(arrayOfArrays.size, 2, 'arrayOfArrays size is as expected');
+
+    t.throws(() => {new Set(1,2,3)}, TypeError,'throws error if not an array');
+    t.throws(() => {new Set(9)}, TypeError, 'throws error with non-iterable obj')
+    t.throws(() => {new Set({1:2, 3:4})}, 'throws error on object constructor');
+
+    t.end();
+  })
+
+  test(`test ${setType} add method`, (t) => {
+    let mySet = new Set();
+    const startSize = mySet.size;
+
+    t.doesNotThrow(() => {mySet.add('a string')}, 'can add string to the set');
+    t.equal(mySet.size, startSize+1, 'adding value increases size by 1');
+    t.doesNotThrow(() => {mySet.add({})}, 'can add object to a set');
+    t.doesNotThrow(() => {mySet.add(()=>{})}, 'can add function to a set');
+    t.skip('diverging from builtin; cannot add null to a set');
+    t.equal(mySet.add(1), mySet, "add returns the set");
+    t.doesNotThrow(() => {mySet.add(1)}, 'can call add with only 1 argument');
+    t.doesNotThrow(() => {mySet.add(2,3,4)}, 'can call add with more than 1 argument');
+    t.ok(mySet.has(2) && !(mySet.has(3)), 'add ignores arguments after 1st arg');
+
+    t.end();
+  });
+
+  test(`test ${setType} has method`, (assert) => {
+    let mySet = new Set();
+    mySet.add(0);
+    mySet.add(1);
+    assert.equal(mySet.has(1), true, 'has(value) returns true when set has value');
+    assert.equal(mySet.has(5), false, 'has(value) returns false for a nonexistent key');
+    assert.equal(mySet.has({}), false, 'has(value) returns false for nonidentical keys');
```

Comment:
Let's add an object `{}` to `mySet` before this assertion, so that we can confirm that it still returns false (since the two `{}`s aren't identical). Would also be good to make sure that it returns true if we give it a reference to an object that's in the set.